module-platform/rt1051/src/disk_emmc.cpp

```diff
@@ -209,4 +211,48 @@ namespace purefs::blkdev
         return kStatus_Success;
     }
 
+    auto disk_emmc::get_card_manufacturer() const -> std::string
+    {
+        const std::uint8_t manufacturerId = mmcCard->cid.manufacturerID;
+        const auto manufacturer           = mmcManufacturersMap.find(manufacturerId);
+        if (manufacturer != mmcManufacturersMap.end()) {
+            return manufacturer->second;
+        }
+        return "unknown (0x" + utils::byteToHex(manufacturerId) + ")";
+    }
+
+    auto disk_emmc::log_emmc_info() const -> void
```

Comment:
How about?
```suggestion
    auto disk_emmc::get_emmc_info_str() const -> std::string
```
Or you can split it into two functions, one for obtaining info structure(you can create custom one containing all the necessary info), and another one for converting it to the human-readable string ;)

Reply:
Please take a look, I've split it into two functions, one returning custom info structure and second one printing all the required values to the log file